r2/r2/controllers/api.py

```diff
@@ -4100,14 +4100,21 @@ def POST_selectflair(self, form, jquery, user, link, flair_template_id,
 
             # XXX: gross UI code
             # Push some client-side updates back to the browser.
+
+            # remove "linkflair" and "linkflair-css_class" classes if they exist
+            jquery('.id-%s' % link._fullname).remove_link_flair_class()
+            # remove the flair itself
             jquery('.id-%s .entry .linkflairlabel' % link._fullname).remove()
             title_path = '.id-%s .entry > .title > .title' % link._fullname
 
             # TODO: move this to a template
             if flair_template:
+                # reinitiate the classes
```

Comment:
nit: I don't think any of these comments are necessary, it's pretty easy to understand what's happening just looking at the code.